numpy/linalg/tests/test_linalg.py

```diff
@@ -845,6 +846,108 @@ def test_empty(self):
         assert_equal(norm(array([], dtype=self.dt)), 0.0)
         assert_equal(norm(atleast_2d(array([], dtype=self.dt))), 0.0)
 
+    def test_vector_types(self):
+        a = np.array([1, 0, 1])
+
+        exact_types = [
+            np.dtype(_).type for _ in np.typecodes['AllInteger']
+        ]
+
+        inexact_types = [
+            np.dtype(_).type for _ in np.typecodes['AllFloat']
+        ]
+
+        all_types = exact_types + inexact_types
+
+        for each_inexact_types in all_types:
+            at = a.astype(each_inexact_types)
+
+            n = norm(at, -np.inf)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 0.0)
+
+            with warnings.catch_warnings() as w:
+                warnings.simplefilter("ignore", RuntimeWarning)
+                n = norm(at, -1)
+                assert_(issubclass(n.dtype.type, np.inexact))
+                assert_almost_equal(n, 0.0)
+
+            n = norm(at, 0)
+            assert_equal(n.dtype.type, np.dtype(int).type)
```

Comment:
I would think that on a 32-bit platform `int` would just be a 32-bit signed int. Perhaps it is, but apparently we resolve it as a 64-bit signed int still. Not sure what can be done about this, but it would be nice to address somehow.